autoscaling/autoscaling.go

```diff
@@ -79,13 +79,7 @@ func (as *AutoScaling) query(params map[string]string, resp interface{}) error {
 
 	hreq.Header.Set("Content-Type", "application/x-www-form-urlencoded; param=value")
 
-	token := as.Auth.Token()
-	if token != "" {
-		hreq.Header.Set("X-Amz-Security-Token", token)
```

Comment:
I moved it inside of `V4Signer.Sign`. I wanted to prevent a race where maybe the token+other credentials are possibly updated after reading the token here but prior to the rest of the signing.